modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PrepareServiceImpl.java

```diff
@@ -316,17 +317,19 @@ private static CacheKey createCacheKey(ParsedResult parsedResult, PlanningContex
 
     private ResultSetMetadata resultSetMetadata(
             RelDataType rowType,
-            @Nullable List<List<String>> origins
+            @Nullable List<List<String>> origins,
+            List<String> aliases
     ) {
         return new LazyResultSetMetadata(
                 () -> {
                     List<ColumnMetadata> fieldsMeta = new ArrayList<>(rowType.getFieldCount());
 
                     for (int i = 0; i < rowType.getFieldCount(); ++i) {
                         RelDataTypeField fld = rowType.getFieldList().get(i);
+                        String alias = aliases.size() > i ? aliases.get(i) : null;
```

Comment:
```suggestion
                        String alias = aliases.size() > i ? aliases.get(i) : fld.getName();
```
and use below `alias`

Reply:
aliases can contain null`s for now, seems it would not become match clear, as i can see.

Reply:
it's not clear for both cases, but suggested one it's a little bit simple